packages/atomic/src/components/atomic-did-you-mean/atomic-did-you-mean.tsx

```diff
@@ -53,11 +50,11 @@ export class AtomicDidYouMean implements InitializableComponent {
 
   private renderAutomaticallyCorrected() {
     const noResults = this.strings.withQuery(
-      'noResultsFor',
+      'no-results-for',
```

Comment:
**Kebab Case vs. Camel Case**
Why does the switch to kebab case make case-insensitive translation possible? Is it because it is not possible to call `.toLowerCase` on registered keys, so we register them as lowercase?

If so, does it also mean that someone registering their own keys to i18n should not add keys with uppercase characters?

**More generally**
I just want to be clear on why case-insensitivity is a requirement? @olamothe mentioned facet values in the scrum being re-indexed with a different case (aka dynamic strings in the page).

Should these values be localized client-side though? Or should they be returned localized by the search api like with localized results?

Also, specifically for facet values, isn't there the value caption option that's needed for facet search to work? Meaning someone should not use i18n to localize facet values?

Considering these things then, is case-insensitivity a true requirement of i18n? Or do dynamic strings need to be managed differently anyway, and i18n would only manage static/client-side strings, which can be case sensitive.

Reply:
There's a bit of the answer in the description of the PR regarding explored solutions and explanation of this current solution. Basically i18n is case **sensitive** and it's not easily modifiable (would take an override of the t function & we would have to change all our keys to lowercase, not readable).

> Why does the switch to kebab case make case-insensitive translation possible?

All the strings we offer are now standardized to kebab-case, this make it less error prone for us & customers than camelCase if they want to override strings. The reason we wanted to allow case insensitivity is to reduce potential errors.

> If so, does it also mean that someone registering their own keys to i18n should not add keys with uppercase characters?

Our strings are standardized, if customers want to add and use their own strings, they can use the casing they choose, we can't really enforce it.

The facet value captions will be worked on in another story. Here is my idea as of now:
- Namespaces will be leveraged https://www.i18next.com/principles/namespaces
- We can use the facet ids for the namespace names
- The facet values will call i18n before being displayed (we can use toLowercase here if we want to be careful of the index values changing)
- Once we get namespaces, we can fetch the "value captions" from i18n for the facet when we send the Facet Search request

Reply:
Additional detail for facet values, let's say I have a facet with the id "author" and the value "amoreau"
The client could override it this way (subject to change a bit):
```
i18next.addResourceBundle('en', 'facet-author', {
  amoreau: 'Alex Moreau'
  // rest of the value captions
});
```

Reply:
Ok, thanks for the clarifications. Would it be correct to say that following this PR:

> i18n is still case sensitive, but the keys are all lowercase to reduce confusion

For the facet values, it looks like a good plan,

Reply:
Yes exactly lowercase with `-` 👍